engine/multisig/src/client/common.rs

```diff
@@ -34,6 +34,7 @@ pub type PublicKey<C> = <<C as ChainSigning>::CryptoScheme as CryptoScheme>::Pub
 pub type SigningPayload<C> = <<C as ChainSigning>::CryptoScheme as CryptoScheme>::SigningPayload;
 pub type Signature<C> = <<C as ChainSigning>::CryptoScheme as CryptoScheme>::Signature;
 
+#[allow(clippy::manual_non_exhaustive)]
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
```

Comment:
Instead of think it would better to actually use the attribute, as the lint suggests: https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute

Reply:
I don't mind changing it but my thinking was that the purpose of the attribute is not relevant to what is being done here, so it felt better to keep the comment but ignore the lint.

The comment here says it's so that you can't use the struct literal to construct the value. 

The attribute is related to public apis and the assumption that the exposed values won't change.

Reply:
I feel it does what we want, but I'm happy for you to leave it.